pkg/store/prometheus.go

```diff
@@ -496,9 +499,46 @@ func (p *PrometheusStore) LabelValues(ctx context.Context, r *storepb.LabelValue
 		return &storepb.LabelValuesResponse{Values: []string{l}}, nil
 	}
 
-	vals, err := p.client.LabelValuesInGRPC(ctx, p.base, r.Label, nil, r.Start, r.End)
-	if err != nil {
-		return nil, err
+	var (
+		sers []map[string]string
+		err  error
+	)
+
+	vals := []string{}
+	version := p.promVersion()
+
+	if version > "2.24" {
```

Comment:
Let's add `if len(r.Matchers) == 0 {` to this line:

```
if len(r.Matchers) == 0 or version > "2.24" {
...
}
```

Reply:
Nice Thought! 👍 
Addressed